### PR TITLE
Map-style operation

### PR DESCRIPTION
**Pull request by dim**

### CODE REVIEW

File: buffer.go [hunk 1]
```diff
@@ -3,50 +3,49 @@ package extsort
...
 }
 
-func (b *memBuffer) Append(data []byte) {
-	n := len(b.chunks)
-	if n < cap(b.chunks) {
-		b.chunks = b.chunks[:n+1]
+func (b *memBuffer) Append(key, val []byte) {
+	n := len(b.ents)
+	if n < cap(b.ents) {
+		b.ents = b.ents[:n+1]
```
Comment (mxmCherry) [inline buffer.go:17]: Does this make it perform observably faster? I see it's an old code adapted, but I'd better go with normal `append`.

File: extsort.go [hunk 2]
```diff
@@ -131,24 +136,34 @@ func (i *Iterator) next() bool {
...
+func (i *Iterator) Key() []byte {
+	return i.ent.Key()
+}
+
+// Value returns the value at the current cursor position.
+func (i *Iterator) Value() []byte {
+	return i.ent.Val()
+}
+
+// Data is an alias for Key, retained for compatibility.
```
Comment (mxmCherry) [inline extsort.go:164]: Nit: I'd not mention anything about "compatibility" here as it kinda implies that this method should not be used in modern code. But this is more about indicating intent - if sorter is used for plain values (`Append([]byte)`) it makes sense to refer to these plain values as `Data()`.

**Review by mxmCherry — APPROVED** [review]

File: buffer.go (hunk 1 above)
Comment (dim) [reply buffer.go:17]: it is to reuse existing `entry` buffers

Comment (dim) [reply buffer.go:17]: basically, if there is an `entry` already stored at position `n+1` it will have an `entry.data` that we can reuse too

**Pull request merged**